src/NuGet.Core/NuGet.Packaging/Signing/Utility/SignatureUtility.cs

```diff
@@ -656,6 +652,26 @@ private static DerSequenceReader CreateDerSequenceReader(CryptographicAttributeO
             return new DerSequenceReader(attribute.Values[0].RawData);
         }
 
+
+        private static byte[] FromHexStringToByteArray(string thumbprint)
+        {
+            byte[] hash = new byte[thumbprint.Length / 2];
+            for (int i = 0; i < hash.Length; i++)
+            {
+                string byteString = thumbprint.Substring(i * 2, 2);
+                if (byte.TryParse(byteString, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out byte hashByte))
+                {
+                    hash[i] = hashByte;
+                }
+                else
+                {
+                    //If parsing fails, return the hash byte array. The mismatch will be processed by throwing SignatureException with SigningCertificateCertificateNotFound
+                    return hash;
+                }
```

Comment:
I appreciate the defensive programming with `TryParse(...)`; however, I don't think we should return a partial hash if `TryParse(...)` fails and leave it to chance for the actual vs. expected hash comparison to fail.

I feel that if `X509Certificate2.Thumbprint` contains invalid hexadecimal, there's a much deeper problem here that should fail verification.

How about if this `if...else` statement is simplified to:
```C#
    hash[i] = byte.Parse(byteString, NumberStyles.HexNumber, CultureInfo.InvariantCulture);
```

Reply:
Thanks for reviewing!
Yeah, I feel the X509Certificate2.Thumbprint should be reliable.
Shall I just let the Parse throw FormatException?
Or I should wrap it with our own exception?